cmd/boulder/main.go

```diff
@@ -53,12 +54,38 @@ func init() {
 		}
 		os.Args = os.Args[1:]
 		subcommand()
-	})
+	}, nil)
 	cmd.RegisterCommand("--list", func() {
 		for _, c := range cmd.AvailableCommands() {
-			if c != "boulder" && c != "--list" {
+			if c != "boulder" && c != "--list" && c != "validate" {
 				fmt.Println(c)
 			}
 		}
-	})
+	}, nil)
+	cmd.RegisterCommand("validate", func() {
```

Comment:
I'd still love to have `validate` and `list` be implemented directly inside `main`, rather than registered as commands which then have to be special-cased by `list`. But this is definitely the right thing to do for now, and I'd like to just file a bug to investigate refactoring these pseudo-commands in the future.

Reply:
Leaving TODO comment with #6763.